lib/practice_exercises.rb

```diff
@@ -1,13 +1,35 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n), the loop requires n operations depending on the length of the string.
+# Space complexity: O(1), the number of varibles tracked does not change regardless of input size.
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  string.gsub!(/[^a-zA-Z0-9]/, "")
+  first_index = 0
+  last_index = string.length - 1
+
+  while first_index < last_index
+    return false if string[first_index].downcase != string[last_index].downcase
+    first_index += 1
+    last_index -= 1
+  end
+  
+  return true 
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2), there is an each loop (dependent on the length of the shortest string) nested within a times loop (dependent on the length of the strings array).
+# Space Complexity: O(1), the number of varibles tracked does not change regardless of input size.
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word and n being the number of strings.  Otherwise 👍